src/Context.cpp

```diff
@@ -42,15 +41,14 @@ Context::~Context() {
     spdlog::shutdown();
 }
 
-std::shared_ptr<Context> Context::CreateInstance(const std::string name, const std::string shortName,
-                                                 const std::string configFilePath,
-                                                 const std::vector<std::string>& archivePaths,
-                                                 const std::unordered_set<uint32_t>& validHashes,
-                                                 uint32_t reservedThreadCount, AudioSettings audioSettings) {
+std::shared_ptr<Context>
```

Comment:
We can check for missing objects in this function and return a nullptr for the Context. We could then have the game check for existence (maybe create a C bridge that returns a book?). If not exist then it's on the game to exit itself since the game will always control `main()`

Reply:
hmm, that doesn't handle the `CreateUninitializedInstance` use case though. if a consumer uses `CreateUninitializedInstance` then calls a few `Init*`s for specific things, then calls `Init` they could still end up with a broken context

Reply:
That's a fair rebuttal, but I think the initial thought is the direction we would want to go. Do we return true/false anywhere on whether or not the `Context` initialized successfully after `Init*`

Reply:
not currently, all the `Init`s are void

https://github.com/Kenix3/libultraship/blob/be703a997b5eb189188ec036223a2238f0c10328/src/Context.h#L40-L41

https://github.com/Kenix3/libultraship/blob/be703a997b5eb189188ec036223a2238f0c10328/src/Context.h#L57-L67

Reply:
Do you think it'd be too kitchen-sinky to correct that and have it return a success/failure state?

Reply:
i don't think that'd be too bad, we could just update

https://github.com/Kenix3/libultraship/blob/be703a997b5eb189188ec036223a2238f0c10328/src/Context.cpp#L79-L91

to be 

```cpp
void Context::Init(const std::vector<std::string>& archivePaths, const std::unordered_set<uint32_t>& validHashes,
                   uint32_t reservedThreadCount, AudioSettings audioSettings) {
    return InitLogging() && InitConfiguration() ...etc;
}
```

do we have a good pattern of returning success/fail somewhere? i'd prefer not to hardcode `0`/`1` if possible

Reply:
https://github.com/Kenix3/libultraship/blob/be703a997b5eb189188ec036223a2238f0c10328/src/public/bridge/resourcebridge.cpp#L11
This here returns a 0 if we can't do the thing. I'd be perfectly happy with bool and true/false.

Reply:
hmm, but that's 0 for size...

i guess true/false is fine - just feels a little odd since it's backwards from exit codes where 0 is good (and returning `false` to mean everything initialized successfully feels odd too)

Reply:
I'm good with that. This is a class initialization. You could have the classes return true/false for `did init` and then the static function swap the return codes since it's the "entry point"

Reply:
cool `true` for init success it is then, i'll get on it